source/write/gridfs.txt

```diff
@@ -0,0 +1,309 @@
+you specify a different name. The library also creates an index on each
+collection to ensure efficient retrieval of the files and related
+metadata. The library creates the GridFS bucket, if it doesn't exist, only when the first write
+operation is performed. The library creates indexes only if they don't exist and when the
+bucket is empty. For more information about
+GridFS indexes, see :manual:`GridFS Indexes </core/gridfs/#gridfs-indexes>`
+in the {+mdb-server+} manual.
+
+When using GridFS to store files, the library splits the files into smaller
+chunks, each represented by a separate document in the ``chunks`` collection.
+It also creates a document in the ``files`` collection that contains
+a file ID, file name, and other file metadata. You can upload the file from
+memory or from a stream. View the following diagram to see how GridFS splits
+the files when uploaded to a bucket:
+
+.. figure:: /includes/figures/GridFS-upload.png
+   :alt: A diagram that shows how GridFS uploads a file to a bucket
+
+When retrieving files, GridFS fetches the metadata from the ``files``
+collection in the specified bucket and uses the information to reconstruct
+the file from documents in the ``chunks`` collection. You can read the file
+into memory or output it to a stream.
+
+.. _gridfs-create-bucket:
+
+Create a GridFS Bucket
+----------------------
+
+To store or retrieve files from GridFS, call the ``MongoDB\Database::selectGridFSBucket()``
+method on your database. This method accesses an existing bucket or creates
+a new bucket if one does not exist.
+
+The following example calls the ``selectGridFSBucket()`` method on the ``db``
+database:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start create bucket
+   :end-before: end create bucket
+
+You can also explicitly create a GridFS bucket by calling the
+constructor for the ``MongoDB\GridFS\Bucket`` class and passing the following
+parameters:
+
+- ``MongoDB\Driver\Manager`` object, which maintains connections between the
+  PHP extension and MongoDB
+- Name of the database, as a string, for which to create the bucket
+
+The following example explicitly constructs a GridFS bucket:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start create bucket explicit
+   :end-before: end create bucket explicit
+
+.. _gridfs-create-custom-bucket:
+
+Customize the Bucket Name
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create or reference a bucket with a custom name other than the default name
+``fs``, pass an options array to the ``selectGridFSBucket()`` method that sets
+the ``bucketName`` option. 
+
+The following example creates a bucket named ``'myCustomBucket'``:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start create custom bucket
+   :end-before: end create custom bucket
+
+.. _gridfs-upload-files:
+
+Upload Files
```

Comment:
It may be helpful to also demonstrate `uploadFromStream()` from the second code examples in [Uploading Files with Writable Streams](https://www.mongodb.com/docs/php-library/current/tutorial/gridfs/#uploading-files-with-writable-streams). That's the suggested API for copying a local file into GridFS.